src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[16];
+    }
+
+    public MyHashMap(int capacity) {
+        if (capacity < 0) {
+            throw new IllegalArgumentException("Illegal initial capacity: " + capacity);
+        }
+        table = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeeded();
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node<>(key, value, null);
+                size++;
+                return;
+            }
+            node = node.next;
+        }
+        table[index] = new Node<>(key, value, null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return (key == null) ? 0 : key.hashCode() & (table.length - 1);
+    }
+
+    private void resizeIfNeeded() {
+        if (size >= table.length * 0.75) {
+            Node<K, V>[] tempArray = table;
+            table = new Node[tempArray.length * 2];
```

Comment:
`oldTable` - более очевидный naming, а на счет`temArray.length * 2` --> `table.lenght * 2`, так сразу понятно, что мы увеличиваем изначальный размер в 2 раза
```suggestion
            Node<K, V>[] oldTable = table;
            table = new Node[table.length * 2];
```